### ISSUE TITLE
Align spec with a few test cases

### ISSUE THREAD

**Issue by bdarcus** [citation-style-language/test-suite#53]

**Comment by Jason-Abbott** [comment]
> Is your implementation open source @Jason-Abbott?

It isn’t currently. I will open-source some or all of it as I finalize some product ideas.

**Comment by Jason-Abbott** [comment]
@adam3smith , @bdarcus following up here after [my learnings](https://github.com/citation-style-language/test-suite/issues/62) on `citation-number` — I would guess those tests that depend on unique and undocumented behaviors of `<key variable="citation-number"/>` belong in this `triage` or similar sub-directory.

I would even argue that these behaviors should never be added to the specification since, although I understand the need for and relative simplicity of the original code change, it breaks the meaning of sort keys. So perhaps an `off-spec` or `citeproc-js` sub-directory …

No strong feeling from me. I’d only like to assist future processor writers or debuggers 🙂

**Comment by Jason-Abbott** [comment]
Yeah, I’d be happy to do a PR. I’ll need to review those `citation-number` tests again since a few don’t require the off-spec behavior … so, some time this week.

**Comment by zepinglee** [comment]
> I would even argue that these behaviors should never be added to the specification since, although I understand the need for and relative simplicity of the original code change, it breaks the meaning of sort keys. So perhaps an `off-spec` or `citeproc-js` sub-directory …

I agree. Those `sort_` tests are confusing to me and I skipped them in CI.

The `citepoc-js` has a local directory (https://github.com/Juris-M/citeproc-js/tree/master/fixtures/local) of tests specific to itself. Perhaps we can move them there?

**Comment by adam3smith** [comment]
I'm still struggling with how to think about things were the test behavior _may_ be desirable but probably shouldn't be spec'd. To take this test/behavior as an example: Frank added the behavior to citeproc-js (and then added a test) based on a user request and to allow for something that would otherwise not be possible, re-interpreting a syntax that is valid according to the specs (sorting by citation-number descending) but (literally) paradoxical. As I mention to Jason above -- a processor would almost certainly be fine without meeting this test. I don't necessarily want to burden the spec with micro-managing behavior like this. At the same time, it seems wasteful to throw out the collected experience represented by tests in this category. That's why I'm suggesting a `recommended` category above. Not sure if that's the right way to go about this, but I also worry that both alternatives I see -- writing every edge case into the specs in great detail or completely deprecating tests of this nature -- have significant downsides. I understand Jason as expressing similar ambivalence.

To move forward without getting bogged down on this, I think we either 
- can table the idea of keeping some off-spec tests as recommended, which would mean just creating a single off-spec directory now and then at a later point revisit a) the idea of recommended, off-spec tests and b) the status of individual off-spec tests or
- agree on the idea of recommended off-spec tests now and create separate subdirectories

I guess wrt moving them to citeproc-js local, I'm not sure what that implies in terms of advice to citeproc developers?

**Comment by Jason-Abbott** [comment]
I hadn't noticed, @zepinglee, that citeproc-js has its own local tests built in this same way. I see sections I don't recognize in some of the tests there so apparently the test harness is also extended there.

Even so, I tend to agree with @adam3smith insofar as the impact of moving tests from here to there is unclear,  an intermediate step of moving them into a subfolder here is safer, until that can be better evaluated.

Personally, my goal is to deliver a processor that behaves as Zotero and other users expect, which could well mean it incorporates citeproc-js idiosyncrasies. That's why I've so far adapted my code to _all_ the tests here.

Seeing all those other local citeproc-js tests adds a wrinkle I'll need to think about. I'm fairly new to this CSL space. I imagine there's a conceptual distinction between off-spec tests here and local tests in citeproc-js, though I sure can't say what it is yet.

**Comment by fbennett** [comment]
If it's at all helpful, documentation on CSL-M idiosyncracies is linked
from the Jurism website.

https://juris-m.github.io/cslm-docs/

Frank
